### ISSUE TITLE
Calculation of information by a server

### ISSUE THREAD

**Issue by gogo25171** [osmandapp/OsmAnd#15488]
Hello, I propose the idea of making a system where you can have a phone without the card files and that its files are stored on a nas, vps, or other server and that the phone retrieves the information from there.
We can even think that all the information is calculated by the nas, vps, etc... And that the telephone or tablet is a man-machine interface.
This will also allow it to be downloaded once on the osmand servers and then it will be the server of the person who takes the relay on all their devices.
Thank you in advance, truthfully yours.

**Comment by pebogufi** [comment]
You can already use online navigation, have a look on navigation options.

**Comment by scaidermern** [comment]
Also online maps and online search are available. So looks like most (if not all?) functionality is already present?

**Comment by gogo25171** [comment]
Even with online navigation you also don't have the route details.

**Comment by tradavyvy** [comment]
My 2 cents:
Off-line is the strength of osmand.
There is no principle problem with on-line (exept in the desert)
But on that front osmand will never win from google.
So i think investing (programmers time) in off-line features is the smarter move.

**Comment by gogo25171** [comment]
I don't think so, because it's always good to have things happening online because the phone's battery will be less drained, and performance will be less used.
And I proposed it is to put the server either thanks to volunteers, or on its own server or on the official servers of Osman.

**Comment by scaidermern** [comment]
> I don't think so, because it's always good to have things happening online because the phone's battery will be less drained, and performance will be less used.

Nope. A phone offline in flight mode will consume less energy than a phone with active network having to retrieve all information via mobile data.

**Comment by gogo25171** [comment]
Hello, you're not entirely wrong, although I think it has to perform a lot more calculations, but with our phones, this might be mitigated. I think that offering the option of cloud computing (NAS, VM, etc.), whether for hosting or processing, helps to conserve phone resources, particularly in terms of battery and storage space.

This way, the phone wouldn't need to provide all that processing power. However, the major drawback would be the development, which could be quite lengthy.

Furthermore, I know that online routing is already possible, but some options—like, for example, knowing the elevation profile after routing—weren't supported at the time. Perhaps this has been fixed since, but since this option is no longer available in the beta version, I haven't been able to retest it (which is rather strange since it's still in the documentation).

https://osmand.net/fr/docs/user/navigation/routing/online-routing/?current-os=ios&

<img width="585" height="1266" alt="Image" src="https://github.com/user-attachments/assets/75dedaaa-8f9e-42f5-a6b9-b6fcb312e128" />
